enhancements/oc/cli-manager.md

```diff
@@ -0,0 +1,231 @@
+of services on OpenShift. We need a mechanism for providing and consuming tools that is simple to add on to as new plugins are
+developed from a variety of sources - and this should be specific for each cluster and available with disconnected installs.
+
+### Goals
+
+* No new form of binary distribution or binary creation will be proposed, because we have an existing structure at Red Hat.
+RPMs or images are the only options, and images must be deployed by the RH pipeline via operators. This proposal is for delivering
+plugins via images, because this will enable offering plugins offline through an existing image registry.
+* Plugin owners must be able to easily distribute their binaries
+* Allow cluster-admins to control which plugins are offered to users
+  * This proposal is not concerned with _which_ plugins will be managed, as that is decided by cluster-admins
+* Controller for registering CRs with an API for listing, extracting and downloading plugins
+  * An API that generates an index that `krew` can consume via its custom index feature
+  * The index and binaries will served from a single route and service within the cluster
+* The `krew` client package will be vendored into `oc` for usage as `oc krew install <plugin>`
+
+### Non-Goals
+
+* The Controller will not build or package plugins
+* No recommending/limiting which plugins are served by the API
+* `krew` will not create or update the Plugin CRs, those will be managed by cluster-admins
+
+## Proposal
+
+Each component wishing to provide customers with their plugins will build and publish images via a trusted image registry
+and create a Plugin CR to provide an image name and the file path within that image for the binaries.
+Clients (i.e. `krew`) will read the index and download binaries from the Controller. The Controller is responsible for building the index from CRs, for pulling the images,
+extracting the binaries, and serving them to clients.
+
+* `krew` and `krew` plugins are upstream projects that Kubernetes users are already familiar with
+* A `krew`-compatible custom index can provide available plugins for a cluster in disconnected environments
+* The index will be served by a Controller with its contents managed by cluster-admins via CRs
+* The binaries will also be served by the Controller that will pull images from a trusted image registry and extract the binaries
+* With `krew index add https://someother-third-party-index` we won't limit users from adding their own index with whatever plugins they want
+* The controller could be optional, since connected environments can use the default `krew` index out-of-the box
+* As `krew` itself is a `kubectl` plugin, it can be invoked using either using `kubectl krew` or `oc krew`
+
+Existing methods of downloading binaries (i.e. the console) will not be affected by this proposal. For the initial implementation, supported plugins will create Plugin CRs. The plugins will be downloaded and installed using `krew`.
```

Comment:
will the console be enabled to serve/show this content?  or you'll have to use the cli?

Reply:
If we go the built-in route, ideally I'd like for console to be able to re-use that. Iow. we'd replace the current container which serves those binaries from cli-artifacts with this mechanism.